test/Microsoft.Azure.ServiceBus.UnitTests/PluginTests.cs

```diff
@@ -152,6 +155,56 @@ public class PluginTests
                 await messageReceiver.CloseAsync();
             }
         }
+
+        [Fact]
+        [DisplayTestMethodName]
+        async Task QueueClientShouldPassPluginsToMessageSession()
+        {
+            var queueClient = new QueueClient(TestUtility.NamespaceConnectionString, TestConstants.SessionNonPartitionedQueueName);
+            try
+            {
+                bool messageReceived = false;
+                var sendReceivePlugin = new SendReceivePlugin();
+                queueClient.RegisterPlugin(sendReceivePlugin);
+
+                var sendMessage = new Message(Encoding.UTF8.GetBytes("Test message"))
+                {
+                    MessageId = Guid.NewGuid().ToString(),
+                    SessionId = Guid.NewGuid().ToString()
+                };
+                await queueClient.SendAsync(sendMessage);
+
+                // Ensure the plugin is called.
+                Assert.True(sendReceivePlugin.MessageBodies.ContainsKey(sendMessage.MessageId));
+
+                queueClient.RegisterSessionHandler(
+                    (session, message, cancellationToken) =>
+                    {
+                        Assert.Equal(sendMessage.SessionId, session.SessionId);
+                        Assert.True(session.RegisteredPlugins.Contains(sendReceivePlugin));
+                        Assert.Equal(sendMessage.Body, message.Body);
+
+                        messageReceived = true;
+                        return Task.CompletedTask;
+                    },
+                    exceptionArgs => Task.CompletedTask);
+
+                for (int i = 0; i < 20; i++)
```

Comment:
Its not exactly a retry. Wanted to wait for around 30-40 seconds before I throw failure. So instead of 10 times delay of 4 seconds, 20 times delay of 2 seconds seems effective. 😄